scripts/firedrake-install

```diff
@@ -76,7 +76,7 @@ class FiredrakeConfiguration(dict):
                     self["options"][o] = args.__dict__[o]
 
     _persistent_options = ["package_manager",
-                           "minimal_petsc", "mpicc", "mpicxx", "mpif90", "mpiexec", "disable_ssh",
+                           "minimal_petsc", "mpicc", "mpicxx", "mpif90", "mpiexec", "mpihome", "disable_ssh",
```

Comment:
Can the mpi* options here be replaced with mpi_dir or similar (which would mirror the PETSc configure argument)?

Reply:
I agree that that would be nicer. But I think this is simpler to implement and `firedrake-install` might well die once this PR is merged.

Reply:
Maybe we can sniff this new value from `args.mpiexec`

Reply:
This will not work as sometimes we pass things like `--mpicc mpicc` (i.e. not a path).